client/lib/posts/post-list-store.js

```diff
@@ -1,371 +1,385 @@
 /**
  * External dependencies
  */
-var debug = require( 'debug' )( 'calypso:posts-list' ),
-	clone = require( 'lodash/lang/clone' ),
-	assign = require( 'lodash/object/assign' ),
-	transform = require( 'lodash/object/transform' ),
-	difference = require( 'lodash/array/difference' ),
-	last = require( 'lodash/array/last' ),
-	max = require( 'lodash/collection/max' );
+import debugLogger from 'debug';
+import clone from 'lodash/lang/clone';
+import assign from 'lodash/object/assign';
+import transform from 'lodash/object/transform';
+import difference from 'lodash/array/difference';
+import last from 'lodash/array/last';
+import max from 'lodash/collection/max';
+import events from 'events/';
 
 /**
  * Internal dependencies
  */
-var Emitter = require( 'lib/mixins/emitter' ),
-	Dispatcher = require( 'dispatcher' ),
-	treeConvert = require( 'lib/tree-convert' )( 'ID' ),
-	PostsStore = require( './posts-store' );
-
-var _defaultQuery = {
-	siteID: false,
-	type: 'post',
-	status: 'publish',
-	orderBy: 'date',
-	order: 'DESC',
-	author: false,
-	search: false,
-	perPage: 20
-};
+import Dispatcher from 'dispatcher';
+import treeConvert from 'lib/tree-convert';
+import PostsStore from './posts-store';
+import PostListCacheStore from './post-list-cache-store';
 
-var _activeList = {
-	postIds: [],
-	errors: [],
-	query: clone( _defaultQuery ),
-	page: 0,
-	nextPageHandle: false,
-	isLastPage: false,
-	isFetchingNextPage: false,
-	isFetchingUpdated: false
-};
+/**
+ * Module Variables
+ */
+const debug = debugLogger( 'calypso:posts-list' );
+const EventEmitter = events.EventEmitter;
 
-var _nextId = 0;
-var PostListStore;
+module.exports = function( id ) {
+	if ( ! id ) {
+		throw new Error( 'must supply a post-list-store id' );
+	}
 
-function queryPosts( options ) {
-	var query = assign( {}, _defaultQuery, options ),
-		cache = require( './post-list-cache-store' ),
-		list;
+	const _defaultQuery = {
+		siteID: false,
+		type: 'post',
+		status: 'publish',
+		orderBy: 'date',
+		order: 'DESC',
+		author: false,
+		search: false,
+		perPage: 20
+	};
+
+	let _activeList = {
+		postIds: [],
+		errors: [],
+		query: clone( _defaultQuery ),
+		page: 0,
+		nextPageHandle: false,
+		isLastPage: false,
+		isFetchingNextPage: false,
+		isFetchingUpdated: false
+	};
+
+	let _nextId = 0;
```

Comment:
A bit mind-boggling to have an ID per query, per ID'd PostListStore :confounded: Should this be incremented globally or per-ListStore? Because `PostListCacheStore` is still global, I'm inclined to think the former.